### ISSUE TITLE
ElementID cannot be null

### ISSUE THREAD

**Issue by jmcgaha** [barrelstrength/craft-sprout-lists#26]
### Description
Just installed the plugin to see if it will fit my needs but I try adding a new list and it gives this error:

```
PDOException: SQLSTATE[23000]: Integrity constraint violation: 1048 Column 'elementId' cannot be null in /Users/me/Sites/site-name/vendor/yiisoft/yii2/db/Command.php:1258
Stack trace:
#0 /Users/me/Sites/site-name/vendor/yiisoft/yii2/db/Command.php(1258): PDOStatement->execute()
#1 /Users/me/Sites/site-name/vendor/yiisoft/yii2/db/Command.php(1075): yii\db\Command->internalExecute('INSERT INTO `sp...')
#2 /Users/me/Sites/site-name/vendor/yiisoft/yii2/db/Schema.php(433): yii\db\Command->execute()
#3 /Users/me/Sites/site-name/vendor/yiisoft/yii2/db/ActiveRecord.php(549): yii\db\Schema->insert('{{%sproutlists_...', Array)
#4 /Users/me/Sites/site-name/vendor/yiisoft/yii2/db/ActiveRecord.php(515): yii\db\ActiveRecord->insertInternal(NULL)
#5 /Users/me/Sites/site-name/vendor/yiisoft/yii2/db/BaseActiveRecord.php(670): yii\db\ActiveRecord->insert(false, NULL)
#6 /Users/me/Sites/site-name/vendor/barrelstrength/sprout-lists/src/elements/Lists.php(212): yii\db\BaseActiveRecord->save(false)
#7 /Users/me/Sites/site-name/vendor/craftcms/cms/src/services/Elements.php(498): barrelstrength\sproutlists\elements\Lists->afterSave(true)
#8 /Users/me/Sites/site-name/vendor/barrelstrength/sprout-lists/src/listtypes/SubscriberListType.php(45): craft\services\Elements->saveElement(Object(barrelstrength\sproutlists\elements\Lists))
#9 /Users/me/Sites/site-name/vendor/barrelstrength/sprout-lists/src/controllers/ListsController.php(91): barrelstrength\sproutlists\listtypes\SubscriberListType->saveList(Object(barrelstrength\sproutlists\elements\Lists))
#10 [internal function]: barrelstrength\sproutlists\controllers\ListsController->actionSaveList()
#11 /Users/me/Sites/site-name/vendor/yiisoft/yii2/base/InlineAction.php(57): call_user_func_array(Array, Array)
#12 /Users/me/Sites/site-name/vendor/yiisoft/yii2/base/Controller.php(157): yii\base\InlineAction->runWithParams(Array)
#13 /Users/me/Sites/site-name/vendor/craftcms/cms/src/web/Controller.php(104): yii\base\Controller->runAction('save-list', Array)
#14 /Users/me/Sites/site-name/vendor/yiisoft/yii2/base/Module.php(528): craft\web\Controller->runAction('save-list', Array)
#15 /Users/me/Sites/site-name/vendor/craftcms/cms/src/web/Application.php(282): yii\base\Module->runAction('sprout-lists/li...', Array)
#16 /Users/me/Sites/site-name/vendor/craftcms/cms/src/web/Application.php(542): craft\web\Application->runAction('sprout-lists/li...', Array)
#17 /Users/me/Sites/site-name/vendor/craftcms/cms/src/web/Application.php(266): craft\web\Application->_processActionRequest(Object(craft\web\Request))
#18 /Users/me/Sites/site-name/vendor/yiisoft/yii2/base/Application.php(386): craft\web\Application->handleRequest(Object(craft\web\Request))
#19 /Users/me/Sites/site-name/public/index.php(21): yii\base\Application->run()
#20 {main}

Next yii\db\IntegrityException: SQLSTATE[23000]: Integrity constraint violation: 1048 Column 'elementId' cannot be null
The SQL being executed was: INSERT INTO `sproutlists_lists` (`id`, `elementId`, `type`, `name`, `handle`, `dateCreated`, `uid`, `dateUpdated`) VALUES (21511, NULL, 'barrelstrength\\sproutlists\\listtypes\\SubscriberListType', 'Test2', 'test2', '2018-12-04 20:55:29', '56b54fd5-168f-4cac-9903-a00605bbb897', '2018-12-04 20:55:29') in /Users/me/Sites/site-name/vendor/yiisoft/yii2/db/Schema.php:664
Stack trace:
#0 /Users/me/Sites/site-name/vendor/yiisoft/yii2/db/Command.php(1263): yii\db\Schema->convertException(Object(PDOException), 'INSERT INTO `sp...')
#1 /Users/me/Sites/site-name/vendor/yiisoft/yii2/db/Command.php(1075): yii\db\Command->internalExecute('INSERT INTO `sp...')
#2 /Users/me/Sites/site-name/vendor/yiisoft/yii2/db/Schema.php(433): yii\db\Command->execute()
#3 /Users/me/Sites/site-name/vendor/yiisoft/yii2/db/ActiveRecord.php(549): yii\db\Schema->insert('{{%sproutlists_...', Array)
#4 /Users/me/Sites/site-name/vendor/yiisoft/yii2/db/ActiveRecord.php(515): yii\db\ActiveRecord->insertInternal(NULL)
#5 /Users/me/Sites/site-name/vendor/yiisoft/yii2/db/BaseActiveRecord.php(670): yii\db\ActiveRecord->insert(false, NULL)
#6 /Users/me/Sites/site-name/vendor/barrelstrength/sprout-lists/src/elements/Lists.php(212): yii\db\BaseActiveRecord->save(false)
#7 /Users/me/Sites/site-name/vendor/craftcms/cms/src/services/Elements.php(498): barrelstrength\sproutlists\elements\Lists->afterSave(true)
#8 /Users/me/Sites/site-name/vendor/barrelstrength/sprout-lists/src/listtypes/SubscriberListType.php(45): craft\services\Elements->saveElement(Object(barrelstrength\sproutlists\elements\Lists))
#9 /Users/me/Sites/site-name/vendor/barrelstrength/sprout-lists/src/controllers/ListsController.php(91): barrelstrength\sproutlists\listtypes\SubscriberListType->saveList(Object(barrelstrength\sproutlists\elements\Lists))
#10 [internal function]: barrelstrength\sproutlists\controllers\ListsController->actionSaveList()
#11 /Users/me/Sites/site-name/vendor/yiisoft/yii2/base/InlineAction.php(57): call_user_func_array(Array, Array)
#12 /Users/me/Sites/site-name/vendor/yiisoft/yii2/base/Controller.php(157): yii\base\InlineAction->runWithParams(Array)
#13 /Users/me/Sites/site-name/vendor/craftcms/cms/src/web/Controller.php(104): yii\base\Controller->runAction('save-list', Array)
#14 /Users/me/Sites/site-name/vendor/yiisoft/yii2/base/Module.php(528): craft\web\Controller->runAction('save-list', Array)
#15 /Users/me/Sites/site-name/vendor/craftcms/cms/src/web/Application.php(282): yii\base\Module->runAction('sprout-lists/li...', Array)
#16 /Users/me/Sites/site-name/vendor/craftcms/cms/src/web/Application.php(542): craft\web\Application->runAction('sprout-lists/li...', Array)
#17 /Users/me/Sites/site-name/vendor/craftcms/cms/src/web/Application.php(266): craft\web\Application->_processActionRequest(Object(craft\web\Request))
#18 /Users/me/Sites/site-name/vendor/yiisoft/yii2/base/Application.php(386): craft\web\Application->handleRequest(Object(craft\web\Request))
#19 /Users/me/Sites/site-name/public/index.php(21): yii\base\Application->run()
#20 {main}
Additional Information:
Array
(
    [0] => 23000
    [1] => 1048
    [2] => Column 'elementId' cannot be null
)

Copy Stacktrace Search Stackoverflow Search Google Exception
Integrity constraint violation – yii\db\IntegrityException
SQLSTATE[23000]: Integrity constraint violation: 1048 Column 'elementId' cannot be null
The SQL being executed was: INSERT INTO `sproutlists_lists` (`id`, `elementId`, `type`, `name`, `handle`, `dateCreated`, `uid`, `dateUpdated`) VALUES (21511, NULL, 'barrelstrength\\sproutlists\\listtypes\\SubscriberListType', 'Test2', 'test2', '2018-12-04 20:55:29', '56b54fd5-168f-4cac-9903-a00605bbb897', '2018-12-04 20:55:29')
Error Info: Array
(
    [0] => 23000
    [1] => 1048
    [2] => Column 'elementId' cannot be null
)
↵
Caused by: PDOException
SQLSTATE[23000]: Integrity constraint violation: 1048 Column 'elementId' cannot be null
in /Users/me/Sites/site-name/vendor/yiisoft/yii2/db/Command.php at line 1258
```

### Steps to reproduce

1. Install plugin
2. Click Add List
3. Enter name
4. Hit submit

### Additional info

- Craft version: 3.0.33
- PHP version: 7.2
- Database driver & version: mysql
- Plugins & versions: beta 12 and dev-develop

**Comment by pdaleramirez** [comment]
I fixed this bug in these commit https://github.com/barrelstrength/craft-sprout-lists/commit/1fb6efaa9b9cb067e42f691c0c90cdaef402ea6a

To make this work uninstall your sprout lists plugin and run install again.

Tests:
- Adding new list in control panel
- Adding subscriber to the newly created list
- Subscribing to the new list via twig template with
- Subscribing to the new list via twig template without elementId

**Comment by BenParizek** [comment]
@jmcgaha We should have an official release out in the next couple days that solves this. We ended up doing it slightly different than the above commit so probably best to wait.

**Comment by BenParizek** [comment]
@jmcgaha This should be resolved in v1.0.0-beta.14
